processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -298,38 +297,15 @@ else if ( reportErrorOnTargetObject( mapping ) ) {
 
         private void handleDefinedNestedTargetMapping(Set<String> handledTargets) {
 
-            Map<PropertyEntry, MappingOptions> optionsByNestedTarget =
-                method.getMappingOptions().groupByPoppedTargetReferences();
-            for ( Entry<PropertyEntry, MappingOptions> entryByTP : optionsByNestedTarget.entrySet() ) {
+            NestedTargetPropertyMappingHolder holder = new NestedTargetPropertyMappingHolder.Builder()
```

Comment:
So.. sorting them into nested targets and non nested targets first thing would clarify that. Especially if you combine that with working on `Map<String,List<Mapping>`> iso `MappingOptions`.